src/mlpack/bindings/python/mlpack/matrix_utils.py

```diff
@@ -52,10 +52,21 @@ def to_matrix(x, dtype=np.double, copy=False):
       return x.copy("C"), True
     else:
       return x, False
+  elif (isinstance(x, np.ndarray) and x.dtype == dtype and x.flags.f_contiguous):
+    if copy: # Copy the matrix if required.
+      return np.ndarray(x.shape, buffer=x.flatten(), dtype=x.dtype, order='C').copy("C"), True
+    else:
+      return np.ndarray(x.shape, buffer=x.flatten(), dtype=x.dtype, order='C'), False
   else:
-    if isinstance(x, pd.core.series.Series):
-      x = pd.DataFrame(x)
-    return np.array(x, copy=True, dtype=dtype, order='C'), True
+    if isinstance(x, pd.core.series.Series) or isinstance(x, pd.DataFrame):
+      y = x.values
+      if copy: # Copy the matrix if required.
+        return np.ndarray(y.shape, buffer=y.flatten(), dtype=y.dtype, order='C').copy("C"), True
+      else:
+        return np.ndarray(y.shape, buffer=y.flatten(), dtype=y.dtype, order='C'), False
```

Comment:
Are you sure about the `y.dtype` here?  It's possible that this may not be the desired `dtype`.

Reply:
No, I thing we should use `dtype` only, I am Sorry I was testing Pandas series so I have used `y.dtype` . But i realize `dtype` gives desired result correctly. Should I make another commit to correct this.??

Reply:
Yes, absolutely---we should not commit incorrect code.

Reply:
Resolved `dtype`